chia-protocol/src/message_struct.rs

```diff
@@ -1,45 +1,56 @@
 #[macro_export]
 macro_rules! message_struct {
-    ($name:ident {$($field:ident: $t:ty $(,)? )*}) => {
-        #[cfg_attr(feature = "py-bindings", pyo3::pyclass(get_all, frozen), derive(chia_py_streamable_macro::PyJsonDict, chia_py_streamable_macro::PyStreamable))]
-        #[derive(Streamable, Hash, Debug, Clone, Eq, PartialEq)]
-        #[cfg_attr(fuzzing, derive(arbitrary::Arbitrary))]
-        pub struct $name {
-            $(pub $field: $t),*
-        }
-
-        #[cfg(not(feature = "py-bindings"))]
-        #[allow(clippy::too_many_arguments)]
-        impl $name {
-            pub fn new ( $($field: $t),* ) -> $name {
-                $name { $($field),* }
-            }
-        }
+    ( $name:ident $( { $( $field:ident: $t:ty ),* $(,)? } )? ) => {
+        $crate::streamable_struct!( $name $( { $( $field: $t, )* } )? );
 
         impl ChiaProtocolMessage for $name {
             fn msg_type() -> $crate::ProtocolMessageTypes {
                 $crate::ProtocolMessageTypes::$name
             }
         }
-    }
+    };
 }
 
 #[macro_export]
 macro_rules! streamable_struct {
-    ($name:ident {$($field:ident: $t:ty $(,)? )*}) => {
+    ( $name:ident { $( $field:ident: $t:ty ),+ $(,)? } ) => {
+        $crate::streamable_struct!( impl $name { $( $field: $t, )* } );
+    };
+
+    ( $name:ident {} ) => {
+        $crate::streamable_struct!( impl $name {} );
+
+        impl Default for $name {
+            fn default() -> Self {
+                Self {}
+            }
+        }
+    };
+
+    ( $name:ident ) => {
+        $crate::streamable_struct!( impl $name );
+
+        impl Default for $name {
+            fn default() -> Self {
+                Self
+            }
+        }
+    };
+
+    ( impl $name:ident $( { $( $field:ident: $t:ty ),* $(,)? } )? ) => {
```

Comment:
it there a convention to use the word "impl" to essentially split a macro into two modes?
my instinct would have been to have a separate macro for, maybe called `streamable_struct_impl` or something